pkg/hubble/parser/threefour/parser.go

```diff
@@ -316,13 +316,21 @@ func (p *Parser) resolveEndpoint(ip net.IP, datapathSecurityIdentity uint32) *pb
 	if p.endpointGetter != nil {
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			epIdentity := resolveIdentityConflict(ep.GetIdentity())
-			return &pb.Endpoint{
+			e := &pb.Endpoint{
 				ID:        uint32(ep.GetID()),
 				Identity:  epIdentity,
 				Namespace: ep.GetK8sNamespace(),
 				Labels:    sortAndFilterLabels(p.log, ep.GetLabels(), epIdentity),
 				PodName:   ep.GetK8sPodName(),
 			}
+			if pod := ep.GetPod(); pod != nil {
+				olen := len(pod.GetOwnerReferences())
+				e.Workloads = make([]*pb.Workload, olen)
+				for _, owner := range pod.GetOwnerReferences() {
+					e.Workloads = append(e.Workloads, &pb.Workload{Kind: owner.Kind, Name: owner.Name})
```

Comment:
Appending to a slice pre-allocated via `make(<type>, <size>)` creates a slice of double the size with half of it empty. It either needs an index usage, or be declared as `make(<type>, 0, <size>)`

See the following example:
```
❯ cat main.go
package main

import "fmt"

func main() {
        s1 := make([]int, 4)
        s1 = append(s1, 1)

        s2 := make([]int, 0, 4)
        s2 = append(s2, 2)

        s3 := make([]int, 4)
        s3[0] = 3

        fmt.Println(s1)
        fmt.Println(s2)
        fmt.Println(s3)
}

❯ go run main.go
[0 0 0 0 1]
[2]
[3 0 0 0]
```

This error would also be visible in a unit test

Reply:
Updated